numpy/ma/tests/test_core.py

```diff
@@ -447,6 +447,22 @@ def test_deepcopy(self):
         assert_equal(copied.mask, [0, 0, 0])
         assert_equal(a.mask, [0, 1, 0])
 
+    def test_format(self):
+        a = array([0, 1, 2], mask=[False, True, False])
+        assert_equal(format(a), "[0 -- 2]")
+        assert_equal(format(masked), "--")
+        assert_equal(format(masked, ""), "--")
+
+        # Postponed from PR #15410, perhaps address in the future.
+        # assert_equal(format(masked, " >5"), "   --")
+        # assert_equal(format(masked, " <5"), "--   ")
+
+        # instead we have....
+        with warnings.catch_warnings(record=True) as w:
+            warnings.filterwarnings("always")
+            assert_equal(format(masked, " >5"), "--")
+            assert_equal(len(w), 1, "Expected a FutureWarning for using format_spec with MaskedElement")
```

Comment:
Thanks, can you write this test with a simple `with assert_warns(...):` that should read quite a bit nicer. Otherwise I think we should just go ahead with this.

Reply:
Oh, I see. The construct I used is actually used to test absence of warnings.